transform/optimizer.go

```diff
@@ -1,15 +1,17 @@
-package compiler
+package transform
 
 import (
 	"errors"
+	"fmt"
 
-	"github.com/tinygo-org/tinygo/transform"
+	"github.com/tinygo-org/tinygo/compileopts"
+	"github.com/tinygo-org/tinygo/compiler/ircheck"
 	"tinygo.org/x/go-llvm"
 )
 
 // Run the LLVM optimizer over the module.
 // The inliner can be disabled (if necessary) by passing 0 to the inlinerThreshold.
-func (c *Compiler) Optimize(optLevel, sizeLevel int, inlinerThreshold uint) []error {
+func Optimize(mod llvm.Module, config *compileopts.Config, optLevel, sizeLevel int, inlinerThreshold uint) []error {
```

Comment:
This function does a lot more than just run the LLVM optimizer. It might make sense to update the comment.

Reply:
Oops forgot about this one. I did update the comment in a previous attempt but forgot it this time.